imap_processing/lo/l0/lo_science.py

```diff
@@ -439,3 +439,47 @@ def find_valid_groups(
     ]
     valid_groups = [is_sequential(seq_ctrs) for seq_ctrs in grouped_seq_ctrs]
     return valid_groups
+
+
+def organize_spin_data(dataset: xr.Dataset) -> xr.Dataset:
+    """
+    Organize the spin data for Lo.
+
+    The spin data is spread across 28 fields. This function
+    combines each of those fields into 2D arrays for each
+    epoch and spin.
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        Lo spin data from packets_to_dataset function.
+
+    Returns
+    -------
+    dataset : xr.Dataset
+        Updated dataset with the spin data organized.
+    """
+    # Get the spin data fields
+    spin_fields = [
+        "start_sec_spin",
+        "start_subsec_spin",
+        "esa_neg_dac_spin",
+        "esa_pos_dac_spin",
+        "valid_period_spin",
+        "valid_phase_spin",
+        "period_source_spin",
+    ]
+
+    for spin_field in spin_fields:
+        packet_fields = [f"{spin_field}_{i}" for i in range(1, 29)]
+        # Combine the spin data fields along a new dimension
+        combined_spin_data = xr.concat(
```

Comment:
I think this needs to be reshaped to a 1D array, so the 28 spins recorded in next packet follow the 28 spins in the current packet along the same dimension. Need to confirm with Lo

Reply:
Confirmed with Lo that this needs to be reshaped

Reply:
When you say needs to be reshaped, what do you mean? This is the correct way to do it, or you need to have a different dimensionality?

Reply:
There are 28 spins per packet and the way the code is currently written is those 28 spins have a dimension called "spin" and each packet has an index (epoch). Colin and I had discussed both options and I forgot we settled on instead having the spins all along one dimension. So the 28 spins in each packet would all be lined up along one dimension "spin" and it would be a support variable so it doesn't need to be tied to an epoch in the CDF

Reply:
I am still not sure I totally follow what you are trying to do here at a quick glance, could you add a comment with dimensions / structure of what this becomes? I would have thought you wanted to do `[spin_field1_vals, spin_field2_vals, ...]`, concatenating each spin one after the other. But, this is interleaving the values because you are stacking and then raveling.

I thought you might still have an `epoch` dimension on this variable and each spin would be the extended concatenation like you described. So the variables would have dimensions: `(epoch, spin_number)` So it would be shape: `(npackets, 28)`.

Reply:
Yeah I do want an array of [spin_field1_vals_packet1, spin_field2_vals_packet1, ..., spin_field1_vals_packet2, ....]. It looks like that's what the code is doing based on my unit test. Originally I had dimensions (epoch, spin_number), but I talked to Lo and they want it all in one dimension, so I need to tie it to the overall spin number for the pointing rather than the spin number in the packet or the epoch.